app/src/androidTest/java/com/github/se/project/ui/StudentEndToEnd.kt

```diff
@@ -136,11 +187,122 @@ class EndToEndTest {
     composeTestRule.onNodeWithTag("dropdownAICC").performClick()
     composeTestRule.onNodeWithTag("mapButton").performClick()
     composeTestRule.onNodeWithTag("map").performClick()
-    Thread.sleep(2000)
-    val device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+    Thread.sleep(2000) // Wait for the map to load
+    device.click(device.displayWidth / 2, device.displayHeight / 2)
+    composeTestRule.onNodeWithTag("confirmLocation").performClick()
+    composeTestRule.onNodeWithTag("confirmButton").performClick()
+
+    // Select a tutor
+    composeTestRule.onNodeWithTag("tutorCard_0").performClick()
+    composeTestRule.onNodeWithText("Confirm").performClick()
+    // composeTestRule.onNodeWithTag("confirmButton").performClick()
+    assert(currentLesson!!.title == "End-to-end testing")
+    assert(currentLesson!!.description == "Teach me how to write tests pls")
+    assert(currentLesson!!.subject == Subject.AICC)
+    assert(currentLesson!!.languages == listOf(Language.ENGLISH))
+    assert(currentLesson!!.status == LessonStatus.PENDING_TUTOR_CONFIRMATION)
+
+    // Navigate to the lesson creation screen
+    composeTestRule.onNodeWithText("Find a Tutor").performClick()
+
+    // Create a new lesson
+    composeTestRule.onNodeWithTag("titleField").performTextInput("Help how do I write tests")
+    composeTestRule
+        .onNodeWithTag("DescriptionField")
+        .performTextInput("Teach me how to write tests pls")
+    composeTestRule.onNodeWithTag("DateButton").performClick()
+    onView(withText("OK")).perform(click())
+    composeTestRule.onNodeWithTag("TimeButton").performClick()
```

Comment:
Adding a Thread.sleep(500) can be useful to pass test locally at least in my case.